### DISCUSSION TITLE
How to detect whole head and not just the face

### DISCUSSION THREAD

**Discussion by JosephBotros** [vladmandic/face-api#64] [Q&A] [answered]
Hi, I can detect the face correctly with no problem but what I really need is to detect the whole head from top of the hair to the chin and from left ear to right ear.
I managed to get the top of the hair by removing the background, but as for the ears it's tricky since hair may be covering them.

**Comment by vladmandic** [comment]
using existing models in this library, its limited to face (as most of them are) - and how all libraries that return box for a head work is they just expand face box by some predefined percentage.

**Reply by JosephBotros** [reply]
I tried this approach and it failed, based on 7 different images I found that head width that faceapi get need to be expanded by 105-133%. So this can't be the answer for my problem. Thank you though for your work.

**Comment by vladmandic** [comment]
you can try some different face detection model, perhaps it suits your needs better  
for example <https://github.com/vladmandic/human> library is much more feature rich

**Reply by JosephBotros** [reply]
Thank you for your awesome work, I'm currently testing it and I'm stuck, how do I use it with node-canvas? I tried:
`human.draw.face(canvas,result.face);`
result is the result I got from `await human.detect(tensor, config);`
canvas here is a node-canvas and not HTMLCanvasElement.

Sorry for your trouble.

**Comment by vladmandic** [comment, marked as answer]
@JosephBotros `human.draw.*` methods were designed for **browser** usage only  
i've just published a new version of `Human` 2.1.5 that allows usage in **nodejs** as well  
and new demo for that, see `demos/nodejs/node-canvas.js`

btw, if you have any questions or issues with `Human`, best to post it in it's git repository

**Reply by JosephBotros** [reply]
You are correct, I should have posted there, sorry.

**Reply by JosephBotros** [reply]
You are awesome by the way :D
